doc/dev/backend-plugins.md

```diff
@@ -0,0 +1,342 @@
+# Backend Plugins
+
+## Backend Contract
+
+There exists a _backend contract_ between `libelektra-kdb` and any plugin acting as a backend plugin.
+This contract sets, the order of the phases described above and defines the interaction between a backend plugin and `libelektra-kdb`.
+
+```mermaid
+sequenceDiagram
+    actor user
+    participant kdb as libelektra-kdb
+    participant backend as backend
+    participant other as other-backend
+    participant storage
+    participant validation
+    participant sa as storage-unit-a (e.g. file)
+    participant sb as storage-unit-b (e.g. database)
+
+    user->>+kdb: kdbGet
+    kdb->>backend: init
+    kdb->>other: init
+    kdb->>+backend: resolver
+    backend->>-kdb: "storage-unit-a", update needed
+    kdb->>+other: resolver
+    other->>-kdb: "storage-unit-b", no update needed
+    kdb->>backend: prestorage
+    kdb->>+backend: storage
+    backend->>+storage: &nbsp;
+    storage->>+sa: &nbsp;
+    sa-->>-storage: &nbsp;
+    storage-->>-backend: &nbsp;
+    backend-->>-kdb: &nbsp;
+    kdb->>+backend: poststorage
+    backend->>+validation: &nbsp;
+    validation-->>-backend: &nbsp;
+    backend-->>-kdb: &nbsp;
+    kdb->>kdb: merge backends
+    kdb-->>-user: &nbsp;
+```
+
+```mermaid
+sequenceDiagram
+    actor user
+    participant kdb as libelektra-kdb
+    participant backend as backend
+    participant other as other-backend
+    participant storage
+    participant validation
+    participant sa as storage-unit-a (e.g. file)
+
+    user->>+kdb: kdbSet
+    kdb->>kdb: check backends initialized
+    kdb->>+backend: resolver
+    backend-->>-kdb: "storage-unit-a"
+    critical try to store
+      kdb->>+backend: prestorage
+      backend->>+validation: &nbsp;
+      validation-->>-backend: &nbsp;
+      backend-->>-kdb: &nbsp;
```

Comment:
Weird, the docs for Mermaid said HTML entities are supported. Maybe GitHub only supports the decimal codes not the named ones...

I needed something after the `: `, because Mermaid requires a space after the `:`, but our formatter removes trailing spaces.